### DISCUSSION TITLE
Token without kid but jwks contains key id

### DISCUSSION THREAD

**Discussion by ahoeing** [SDA-SE/sda-dropwizard-commons#2536] [General]
Hi sda-commons developers,

I am currently faced with the following situation.
The issued key has no kid header (but only x5t) defined, but the kid is included in the JKWS, provided by the identity provider.

Due to this line:
https://github.com/SDA-SE/sda-dropwizard-commons/blob/290f65ab1376f0e3e355db2917bc97e43675a144/sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthService.java#L42

I cannot validate the token. Is it mandatory that if the token has no kid, then the key must also have no key id or is it ok to be more flexible here, e.g. check against all known keys is keyid is not given?

Thanks

**Comment by JoergSiebahn** [comment]
As a quick fix, you may configure the key as [PEM](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/config/KeyUriType.java#L6).

Does your token have any other indication of the used key, e.g. an `iss` claim?

I'm not sure about all implications, but at first sight, it feels wrong to try any key for validation. But it could be an option to try any key of a defined issuer.

**Reply by ahoeing** [reply]
Yes, I know that workaround but I need a sustainable solution.
It has an x5t info in it's header. x5t is the fingerprint of the certificate.

**Reply by JoergSiebahn** [reply]
Would it be possible to use that fingerprint to identify the public key directly or in the JWKS?

**Reply by ahoeing** [reply]
yes. We would have to extend the sda-commons implementation to configure the key identifier.

**Comment by JoergSiebahn** [comment]
Solved in https://github.com/SDA-SE/sda-dropwizard-commons/pull/2546
